pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -376,6 +377,34 @@ func GetPipelineConditionStatus(pr *v1alpha1.PipelineRun, state PipelineRunState
 	}
 }
 
+// GetPipelineCompletionTime will return the CompletionTime that the PipelineRun pr should be
+// updated with, based on the status and CompletionTime of the TaskRuns in state.
```

Comment:
It looks like this will also _not_ return a completion time, if any of the TaskRuns aren't yet complete. Is that the intention? It's not in the docstring.

On that note, does it make sense to report the PipelineRun's completion time as the time _we noticed_ the pipeline was done, even if that's some number of seconds after the last TaskRun actually completed?

I could see value in either approach, but I think we lose information about how quickly we reconcile completeness if we always "back-date" completion to the "actually correct" time (i.e., all TaskRuns are done) and not the time the user can first observe completeness.

Reply:
> It looks like this will also not return a completion time, if any of the TaskRuns aren't yet complete. Is that the intention? It's not in the docstring.

Indeed, the function returns a CompletionTime if and only if the PipelineRun is completed (i.e., all tasks have completed). 

See https://github.com/tektoncd/pipeline/issues/1680#issuecomment-566016474

Thanks for this comment, by the way, this is exactly the kind of question I'd like to get clarity on in the associated issue https://github.com/tektoncd/pipeline/issues/1680

I'm not familiar with the code base or the implicit assumptions around the various conditions, I just noticed that the existing behaviour is awkward which lead to my filing https://github.com/tektoncd/pipeline/issues/1680

Feedback around the overall design described in https://github.com/tektoncd/pipeline/issues/1680 would be very welcome.